### PR TITLE
Values for 'mouStart' and 'mouEnd' returned null

### PR DESCRIPTION
**Pull request by ibohalsinghkoijam**

### CODE REVIEW

File: src/components/partnership/partnership.service.ts [hunk 1]
```diff
@@ -383,21 +383,82 @@ export class PartnershipService {
...
+        },
+      ],
+      projectMouEnd: [{ value: 'projectMouEnd' }],
+      projectMouEndPerm: [
+        {
+          read: 'canReadProjectMouEnd',
+          edit: 'canEditProjectMouEnd',
+        },
+      ],
+    });
```
Comment (michaelmarshall) [inline src/components/partnership/partnership.service.ts:440]: It shouldn't be necessary to write new logic when the project service has a readOne function. Please refactor this to call the project service readOne function to get the project mouStart/End dates with permissions.

File: src/components/partnership/partnership.service.ts [hunk 2]
```diff
@@ -383,21 +383,82 @@ export class PartnershipService {
...
+    const readProject = await projectMapPartner.first();
+
     let mouStart = null;
     let mouEnd = null;
 
     // if user has access to project mou and there is no partnership override
-    if (result.mouStart.canRead || result.mouStartOverride.canRead) {
-      mouStart = result.mouStartOverride.value ?? result.mouStart.value;
+    if (
+      readProject?.canReadProjectMouStart ||
```
Comment (michaelmarshall) [inline src/components/partnership/partnership.service.ts:449]: this should be &&, not ||.

File: src/components/partnership/partnership.service.ts [hunk 3]
```diff
@@ -383,21 +383,82 @@ export class PartnershipService {
...
+      result.mouStartOverride.canRead
+    ) {
+      mouStart = result.mouStartOverride.value ?? readProject?.projectMouStart;
     }
-    if (result.mouEnd.canRead || result.mouEndOverride.canRead) {
-      mouEnd = result.mouEndOverride.value ?? result.mouEnd.value;
+    if (readProject?.canReadProjectMouEnd || result.mouEndOverride.canRead) {
```
Comment (michaelmarshall) [inline src/components/partnership/partnership.service.ts:454]: this should be &&, not ||.

File: src/components/partnership/partnership.service.ts [hunk 4]
```diff
@@ -383,21 +383,82 @@ export class PartnershipService {
...
     let mouEnd = null;
 
     // if user has access to project mou and there is no partnership override
-    if (result.mouStart.canRead || result.mouStartOverride.canRead) {
-      mouStart = result.mouStartOverride.value ?? result.mouStart.value;
+    if (
+      readProject?.canReadProjectMouStart ||
+      result.mouStartOverride.canRead
+    ) {
+      mouStart = result.mouStartOverride.value ?? readProject?.projectMouStart;
```
Comment (michaelmarshall) [inline src/components/partnership/partnership.service.ts:452]: this is good, just make sure the value comes from call the project readOne function.

File: src/components/partnership/partnership.service.ts [hunk 5]
```diff
@@ -383,21 +383,82 @@ export class PartnershipService {
...
     }
-    if (result.mouEnd.canRead || result.mouEndOverride.canRead) {
-      mouEnd = result.mouEndOverride.value ?? result.mouEnd.value;
+    if (readProject?.canReadProjectMouEnd || result.mouEndOverride.canRead) {
+      mouEnd = result.mouEndOverride.value ?? readProject?.projectMouEnd;
     }
 
     const canReadMouStart =
-      result.mouStart.canRead || result.mouStartOverride.canRead;
+      readProject?.canReadProjectMouStart || result.mouStartOverride.canRead;
```
Comment (michaelmarshall) [inline src/components/partnership/partnership.service.ts:459]: needs to be &&. You only have permission when you can read both.

File: src/components/partnership/partnership.service.ts [hunk 6]
```diff
@@ -383,21 +383,82 @@ export class PartnershipService {
...
     const canReadMouEnd =
-      result.mouEnd.canRead || result.mouEndOverride.canRead;
+      readProject?.canReadProjectMouEnd || result.mouEndOverride.canRead;
```
Comment (michaelmarshall) [inline src/components/partnership/partnership.service.ts:461]: needs to be &&. You only have permission when you can read both.

File: src/components/partnership/partnership.service.ts (hunk 1 above)
Comment (ibohalsinghkoijam) [reply src/components/partnership/partnership.service.ts:440]: @michaelmarshall , I need to find project(s) mapped with this partnership.

File: src/components/partnership/partnership.service.ts (hunk 3 above)
Comment (ibohalsinghkoijam) [reply src/components/partnership/partnership.service.ts:454]: changed to &&

File: src/components/partnership/partnership.service.ts (hunk 5 above)
Comment (ibohalsinghkoijam) [reply src/components/partnership/partnership.service.ts:459]: changed to &&

File: src/components/partnership/partnership.service.ts (hunk 6 above)
Comment (ibohalsinghkoijam) [reply src/components/partnership/partnership.service.ts:461]: changed to &&

File: src/components/partnership/partnership.service.ts (hunk 2 above)
Comment (ibohalsinghkoijam) [reply src/components/partnership/partnership.service.ts:449]: changed to &&

**Pull request merged**